File: kotlinlib/test/resources/mixed-code-hello-world-kotlin/main/test/src/HelloTest.java
/*
 * Copyright 2020-Present Original lefou/mill-kotlin repository contributors.
 */

package hello.tests;

import hello.JavaHello;
import junit.framework.TestCase;

public class HelloTest extends TestCase {
    public void testSuccess() {
        assertEquals("Hello from Kotlin!", JavaHello.getHelloStringFromKotlin());
        assertEquals("Hello from Java!", hello.KotlinHelloKt.getHelloStringFromJava());
    }

    public void testFailure() {
        assertEquals("Hello from Java!", JavaHello.getHelloStringFromKotlin());
        assertEquals("Hello from Kotlin!", hello.KotlinHelloKt.getHelloStringFromJava());
    }
}

File: libs/kotlinlib/test/resources/hello-kotlin/main/test/src/HelloTest.kt
/*
 * Copyright 2020-Present Original lefou/mill-kotlin repository contributors.
 */

package hello.tests

import hello.getHelloString
import org.junit.Test
import kotlin.test.assertEquals

class HelloTest {
    @Test fun testSuccess() {
        assertEquals("Hello, world!", getHelloString())
    }

    @Test fun testFailure() {
        assertEquals("world!", getHelloString())
    }
}
